presto-main/src/main/java/com/facebook/presto/execution/SqlTaskManager.java

```diff
@@ -419,6 +421,8 @@ public TaskInfo updateTask(
         }
 
         queryContext.setVerboseExceededMemoryLimitErrorsEnabled(isVerboseExceededMemoryLimitErrorsEnabled(session));
+        queryContext.setHeapDumpOnExceededMemoryLimitEnabled(isHeapDumpOnExceededMemoryLimitEnabled(session));
+        queryContext.setHeapDumpFilePath(getHeapDumpFilePath(session));
```

Comment:
can we make the two files immutable and pass them in throw the constructor? We will have a lot of tasks per query and it's not necessary setting them all the time

Reply:
I ran into the same thing. https://github.com/prestodb/presto/blob/master/presto-main/src/main/java/com/facebook/presto/execution/SqlTaskManager.java#L409
QueryContext is constructed from a  `LoadingCache` and when the `LoadingCache` is constructed there is no session. Nest loading caches in fact. I think there are races where the Task and associated QueryContext need to be retrieved before the session is available because messages for a task can arrive before the associated session information does.

Seems like we could simplify by having one setter with two params though?